Casks/chessx.rb

```diff
@@ -9,8 +9,8 @@
   homepage "https://chessx.sourceforge.io/"
 
   livecheck do
-    url "https://sourceforge.net/projects/chessx/files/latest/download"
-    strategy :header_match
+    url "https://sourceforge.net/projects/chessx/rss?path=/chessx"
+    regex(%r{chessx/(\d+(?:\.\d+)+).*?\.dmg}i)
```

Comment:
```suggestion
    regex(%r{url=.*?/chessx[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
```

Matching the version from the directory name works (and accommodates their older setup, where they used a generic `chessx.dmg` file name) but they've used a consistent file format like `chessx-1.2.3.dmg` since ~2013. The only exceptions are the unstable files like `chessx-1.5.7a.dmg` in the `beta` directory, which we don't want to match.

There's something to be said for matching the version from the filename, as this would use a more standard pattern for `SourceForge` checks and is stricter than the regex this PR is proposing. My thinking is that this looser regex would match an unstable version if upstream used a directory name like `1.2.3` with unstable `chessx-1.2.3a.dmg` files inside (instead of something like `beta`).

Reply:
Thanks @samford!  Valid points.  I'll work to be more strict with regexes in the future.

Appreciate the inputs.  Made changes as suggested.